### PR TITLE
Various fixes for mypy, to reduce error count

### PR DESCRIPTION
**Pull request by erezsh**
Some of these "break" the typing interface, so this PR belongs in v1.1

### CODE REVIEW

File: lark/lark.py [hunk 1]
```diff
@@ -540,7 +540,7 @@ def open(cls: Type[_T], grammar_filename: str, rel_to: Optional[str]=None, **opt
             return cls(f, **options)
 
     @classmethod
-    def open_from_package(cls: Type[_T], package: str, grammar_path: str, search_paths: Tuple[str, ...]=("",), **options) -> _T:
+    def open_from_package(cls: Type[_T], package: str, grammar_path: str, search_paths: List[str]=[""], **options) -> _T:
```
Comment (MegaIng) [inline lark/lark.py:543]: Why not `Sequence[str]`?

File: lark/load_grammar.py [hunk 2]
```diff
@@ -804,9 +804,9 @@ class FromPackageLoader:
     """
 
     pkg_name: str
-    search_paths: Tuple[str, ...]
+    search_paths: List[str]
 
-    def __init__(self, pkg_name: str, search_paths: Tuple[str, ...]=("", )) -> None:
+    def __init__(self, pkg_name: str, search_paths: List[str]=[""]) -> None:
```
Comment (MegaIng) [inline lark/load_grammar.py:809]: Same here

**Review by MegaIng — CHANGES_REQUESTED** [review]

File: lark/lark.py (hunk 1 above)
Comment (erezsh) [reply lark/lark.py:543]: I can change it to Sequence. Anything else?

Comment (MegaIng) [reply lark/lark.py:543]: No, looks good otherwise :-)

**Review by MegaIng — APPROVED** [review]

**Comment by erezsh** [comment]
It was surprisingly difficult to install Python 3.6

**Review by MegaIng — APPROVED** [review]
Well, let's hope that Python3.6 wont be used for that much longer anyway. But I am not entirely sure if that is a justified hope...

**Pull request merged**